_code/InSituTree/R/collapseProfiles.R

```diff
@@ -0,0 +1,127 @@
+#' Collapse a hierarchy of cell expression profiles to the root level
+#' as described in the nested list for cell type hierarchy
+#'
+#' @param cth Cell type hierarchy, as a nested list with end point cell types
+#' in full_profiles
+#' @param full_profiles Cell profile matrix, genes x cell types,
+#' include all end point cell types in cth
+#' @param verbose Option to print progress to console
+#' @param print_tree Option to print a structure of the nested lsit
+#'
+#' @return Matrix of collapsed cell expression profiles
+#' @export
+#' @import Matrix
+#'
+#' @examples
+#'
+#' set.seed(123)
+#' test_profiles <- structure(
+#'   rnorm(42, mean = 3, sd = 1),
+#'   .Dim = c(3L, 14L),
+#'   .Dimnames = list(
+#'     c("GeneA", "GeneB", "GeneC"),
+#'     c(
+#'       "B-cell", "endothelial", "fibroblast",
+#'       "macrophage", "mast", "mDC", "monocyte", "neutrophil", "NK",
+#'       "pDC", "plasmablast", "T4", "T8", "Treg"
+#'     )
+#'   )
+#' )
+#'
+#' cth_list <- list(
+#'   structural = c("endothelial", "fibroblast"),
+#'   lymphoid = list(
+#'     `B-lymphoid` = c(
+#'       "B-cell", "pDC",
+#'       "plasmablast"
+#'     ),
+#'     `T-lymphoid` = c(
+#'       "NK", "T4", "T8",
+#'       "Treg"
+#'     )
+#'   ),
+#'   myeloid = c(
+#'     "macrophage", "mast", "mDC", "monocyte",
+#'     "neutrophil"
+#'   )
+#' )
+#'
+#' test <- collapseProfiles(cth = cth_list, full_profiles = test_profiles)
+#' test
+collapseProfiles <- function(cth, full_profiles, verbose = FALSE,
+                             print_tree = FALSE) {
+  # Verify input match
+  if (!all(unlist(cth) %in% colnames(full_profiles))) {
+    stop("The 'cth' list has entries not present
+         in the colnames of 'full_profiles'")
+  }
+
+  # check if there is a list or just the bottom level of the list
+  if (!inherits(celltypes, "list")) {
```

Comment:
`celltypes` is not an input argument, it should be `cth` instead.

Reply:
Good catch, changed